tests/__snapshots__/json_ld.test.ts.snap

```diff
@@ -0,0 +1,226 @@
+export const snapshot = {};
+
+snapshot[`json_ld plugin 1`] = `
+{
+  formats: [
+    {
+      engines: 0,
+      ext: ".page.toml",
+      loader: [AsyncFunction: toml],
+      pageType: "page",
+    },
+    {
+      engines: 1,
+      ext: ".page.ts",
+      loader: [AsyncFunction: module],
+      pageType: "page",
+    },
+    {
+      engines: 1,
+      ext: ".page.js",
+      loader: [AsyncFunction: module],
+      pageType: "page",
+    },
+    {
+      engines: 0,
+      ext: ".page.jsonc",
+      loader: [AsyncFunction: json],
+      pageType: "page",
+    },
+    {
+      engines: 0,
+      ext: ".page.json",
+      loader: [AsyncFunction: json],
+      pageType: "page",
+    },
+    {
+      dataLoader: [AsyncFunction: json],
+      engines: 0,
+      ext: ".json",
+      loader: [AsyncFunction: json],
+    },
+    {
+      dataLoader: [AsyncFunction: json],
+      engines: 0,
+      ext: ".jsonc",
+      loader: [AsyncFunction: json],
+    },
+    {
+      engines: 1,
+      ext: ".md",
+      loader: [AsyncFunction: text],
+      pageType: "page",
+    },
+    {
+      engines: 1,
+      ext: ".markdown",
+      loader: [AsyncFunction: text],
+      pageType: "page",
+    },
+    {
+      dataLoader: [AsyncFunction: module],
+      engines: 1,
+      ext: ".js",
+      loader: [AsyncFunction: module],
+    },
+    {
+      dataLoader: [AsyncFunction: module],
+      engines: 1,
+      ext: ".ts",
+      loader: [AsyncFunction: module],
+    },
+    {
+      engines: 1,
+      ext: ".vento",
+      loader: [AsyncFunction: text],
+      pageType: "page",
+    },
+    {
+      engines: 1,
+      ext: ".vto",
+      loader: [AsyncFunction: text],
+      pageType: "page",
+    },
+    {
+      dataLoader: [AsyncFunction: toml],
+      engines: 0,
+      ext: ".toml",
+      loader: [AsyncFunction: toml],
+    },
+    {
+      dataLoader: [AsyncFunction: yaml],
+      engines: 0,
+      ext: ".yaml",
+      loader: [AsyncFunction: yaml],
+      pageType: "page",
+    },
+    {
+      dataLoader: [AsyncFunction: yaml],
+      engines: 0,
+      ext: ".yml",
+      loader: [AsyncFunction: yaml],
+      pageType: "page",
+    },
+  ],
+  src: [
+    "/",
+    "/page-1.md",
+    "/page-2.vto",
+  ],
+}
+`;
+
+snapshot[`json_ld plugin 2`] = `[]`;
+
+snapshot[`json_ld plugin 3`] = `
+[
+  {
+    content: '<!DOCTYPE html>
+<html><head><script type="application/ld+json">{"@context":"https://schema.org","@type":"WebSite","url":"http://localhost/","name":"Title from page data","inLanguage":"gl","publisher":{"@type":"Organization","name":"Title from page data","logo":{"@type":"ImageObject","url":"./use-cover-as-meta-image.png"}},"image":"http://localhost/my-image.png","keywords":["one","two"]}</script>
```

Comment:
It this correct? `"url":"./use-cover-as-meta-image.png"` is a relative path, I don't know if this is allowed.

Reply:
https://github.com/lumeland/lume/blob/cc31f883ddeb14e07957de8966f92b58d2becc3c/core/site.ts#L745-L752
`Site.url` doesn’t handle paths starting with ./ or ../.

Reply:
Yeah. These urls must be resolved from the page they were generated.

probably this should work:

```js
// Get the absolute URL of the page
const pageUrl = site.url(page.data.url);

// Get the absolute URL of the link
const fullUrl = new URL(url, pageUrl);
```